guides/plugins/plugins/storefront/using-a-modal-window.md

```diff
@@ -0,0 +1,284 @@
+# Using a modal window
+
+## Overview
+
+This guide explains how you can use a modal window in your plugin.
```

Comment:
The guide itself looks good so far. 👍 

I am not sure if we already have another guide or explanation regarding a modal window, but here we only describe the "advanced" or programmatic way of doing this.

There are two more ways we could maybe describe here, and then in an "advanced" section we could describe the stuff you have already prepared.

### Create a modal manually from the DOM using Boostrap

The modal inside the shopware storefront is using the Bootstrap modal. More info: https://getbootstrap.com/docs/4.0/components/modal/#live-demo

```html
<!-- Button trigger modal -->
<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModal">
  Launch demo modal
</button>

<!-- Modal -->
<div class="modal fade" id="exampleModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
  <div class="modal-dialog" role="document">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title" id="exampleModalLabel">Modal title</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        ...
      </div>
      <div class="modal-footer">
        <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
        <button type="button" class="btn btn-primary">Save changes</button>
      </div>
    </div>
  </div>
</div>
```

### Create a modal using Ajax Modal Utility

When setting a `data-url` in addition to `data-toggle="modal"` shopware automatically uses the `PseudoModalUtil` and the pseudo modal template from the `base.html.twig` to render a modal:

```html
<button 
    class="btn"
    data-toggle="modal"
    data-url="https://example.org/ajax-url">
```

**HANDS UP: This does not work when the trigger selector is being changed via JavaScript, e.g. because of an ajax call which replaces the content**

### Advanced / manual using Pseudo Modal Utility

...

Reply:
thanks @tobiasberge  - just added the missing parts